dashboard/src/shared/PackageRepositoriesService.ts

```diff
@@ -0,0 +1,252 @@
+// SPDX-License-Identifier: Apache-2.0
+
+import { Any } from "gen/google/protobuf/any";
+import { Context } from "gen/kubeappsapis/core/packages/v1alpha1/packages";
+import {
+  AddPackageRepositoryRequest,
+  DeletePackageRepositoryResponse,
+  DockerCredentials,
+  GetPackageRepositoryDetailResponse,
+  GetPackageRepositorySummariesResponse,
+  OpaqueCredentials,
+  PackageRepositoryAuth,
+  PackageRepositoryReference,
+  PackageRepositoryTlsConfig,
+  SecretKeyReference,
+  SshCredentials,
+  TlsCertKey,
+  UpdatePackageRepositoryRequest,
+  UsernamePassword,
+} from "gen/kubeappsapis/core/packages/v1alpha1/repositories";
+import {
+  protobufPackage as helmProtobufPackage,
+  RepositoryCustomDetails,
+} from "gen/kubeappsapis/plugins/helm/packages/v1alpha1/helm";
+import KubeappsGrpcClient from "./KubeappsGrpcClient";
+import { IPkgRepoFormData } from "./types";
+import { PluginNames } from "./utils";
+
+export class PackageRepositoriesService {
+  public static coreRepositoriesClient = () =>
+    new KubeappsGrpcClient().getRepositoriesServiceClientImpl();
+  public static helmRepositoriesClient = () =>
+    new KubeappsGrpcClient().getHelmRepositoriesServiceClientImpl();
+
+  public static async getPackageRepositorySummaries(
+    context: Context,
+  ): Promise<GetPackageRepositorySummariesResponse> {
+    return await this.coreRepositoriesClient().GetPackageRepositorySummaries({ context });
+  }
+
+  public static async getPackageRepositoryDetail(
+    packageRepoRef: PackageRepositoryReference,
+  ): Promise<GetPackageRepositoryDetailResponse> {
+    // since the Helm plugin has its own fields (dockerRegistrySecrets, ociRepositories, filterRule, performValidation),
+    // we invoke it directly instead of using kthe core API client.
+    switch (packageRepoRef?.plugin?.name) {
+      case PluginNames.PACKAGES_HELM:
+        return await this.helmRepositoriesClient().GetPackageRepositoryDetail({ packageRepoRef });
+      default:
+        return await this.coreRepositoriesClient().GetPackageRepositoryDetail({ packageRepoRef });
+    }
+  }
+
+  public static async addPackageRepository(
+    cluster: string,
+    namespace: string,
+    request: IPkgRepoFormData,
+    namespaceScoped: boolean,
+  ) {
+    const addPackageRepositoryRequest = this.buildAddOrUpdateRequest(
+      false,
+      cluster,
+      namespace,
+      request,
+      namespaceScoped,
+    );
+
+    // since the Helm plugin has its own fields (dockerRegistrySecrets, ociRepositories, filterRule, performValidation),
+    // we invoke it directly instead of using kthe core API client.
+    switch (request.plugin.name) {
+      case PluginNames.PACKAGES_HELM:
+        return await this.helmRepositoriesClient().AddPackageRepository(
+          addPackageRepositoryRequest,
+        );
+      default:
+        return await this.coreRepositoriesClient().AddPackageRepository(
+          addPackageRepositoryRequest,
+        );
+    }
+  }
```

Comment:
Why having separate calls if the request object has the field `customDetail` is present in `AddPackageRepositoryRequest` and should be processed by the core accordingly?

Reply:
Mmm, I have to re-check. I would say I faced some problems when encoding the customDetails in the javascript client as the core API client didn't know how to encode this message. I'll have a look at it, but, of course, if the core client is able to handle it, this duplication wouldn't be required.

Reply:
It seems it works! (see https://github.com/vmware-tanzu/kubeapps/pull/4954/commits/991c115b22e1c194644e5e04a6b368c4e685a50d). 
However, a more thorough testing is required, but let's see if we can keep it this (simpler) way